src/systemctl/systemctl-edit.c

```diff
@@ -483,6 +483,18 @@ static int trim_edit_markers(const char *path) {
                 strshorten(contents_start, contents_end - contents_start);
 
         contents_start = strstrip(contents_start);
+        if (*contents_start && !endswith(contents_start, "\n")) {
+                if (MALLOC_SIZEOF_SAFE(contents) - (contents_start - contents) - strlen(contents_start) < 2) {
+                        tmp = realloc(contents, size + 1);
+                        if (!tmp)
+                                goto nofix;
```

Comment:
this goto is a bit ugly - just invert the condition check, and add another below